### ISSUE TITLE
make check fails to import build when ganeti already installed

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#226]
Originally reported of Google Code with ID 170.

```
What software version are you running? 

git clone git://git.ganeti.org/ganeti.git
git checkout stable-2.4.2

<b>What distribution are you using?</b>

Debian GNU/Linux sid

<b>What steps will reproduce the problem?</b>
1. apt-get install ganeti2
2. git clone git://git.ganeti.org/ganeti.git
3. cd ganeti
4. git checkout stable-2.4.2
5. autoreconf ; ./configure ; make check

What is the expected output? 

 The test complete.

What do you see instead?

Traceback (most recent call last):
  File "/home/loic/software/ganeti/ganeti/./autotools/build-bash-completion", line 36, in <module>
    from ganeti import build
ImportError: cannot import name build


<b>Please provide any additional information below.</b>

Running with strace -e open shows

open("/usr/lib/pymodules/python2.6/ganeti/build.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/pymodules/python2.6/ganeti/buildmodule.so", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/pymodules/python2.6/ganeti/build.py", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/lib/pymodules/python2.6/ganeti/build.pyc", O_RDONLY) = -1 ENOENT (No such file or directory)

immediately before the traceback. Even though PYTHONPATH=. is set, /usr/lib/pymodules/python2.6 is looked up first and triggers the failure.

```
Originally added on 2011-06-21 10:53:47 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The line
git checkout stable-2.4.2
must read
git checkout stable-2.4

```
Originally added on 2011-06-21 11:03:23 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Can you try building with “PYTHONPATH=. make”? Also, please use “autogen.sh” instead of calling auto* manually.
```
Originally added on 2011-06-21 12:47:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

I tried something slightly different:

apt-get install ganeti2
tar -zxvf ganeti-2.4.2.tar.gz 
cd ganeti-2.4.2
./configure
PYTHONPATH=. make
...
PYTHONPATH=. ./autotools/run-in-tempdir /home/loic/software/ganeti/ganeti-2.4.2/./autotools/build-bash-completion > doc/examples/bash_completion
Traceback (most recent call last):
  File "/home/loic/software/ganeti/ganeti-2.4.2/./autotools/build-bash-completion", line 36, in <module>
    from ganeti import build
ImportError: cannot import name build
make[1]: *** [doc/examples/bash_completion] Error 1
make[1]: *** Deleting file `doc/examples/bash_completion'
make[1]: Leaving directory `/home/loic/software/ganeti/ganeti-2.4.2'
make: *** [all] Error 2
sudo apt-get remove ganeti2
PYTHONPATH=. make
cd . && test -h "ganeti" || { rm -f ganeti && ln -s lib ganeti; }
make  all-am
make[1]: Entering directory `/home/loic/software/ganeti/ganeti-2.4.2'
PYTHONPATH=. ./autotools/run-in-tempdir /home/loic/software/ganeti/ganeti-2.4.2/./autotools/build-bash-completion > doc/examples/bash_completion
sed -f autotools/replace_vars.sed < doc/examples/ganeti.cron.in > doc/examples/ganeti.cron
sed -f autotools/replace_vars.sed < doc/examples/ganeti.initd.in > doc/examples/ganeti.initd
sed -f autotools/replace_vars.sed < doc/examples/ganeti-kvm-poweroff.initd.in > doc/examples/ganeti-kvm-poweroff.initd
sed -f autotools/replace_vars.sed < doc/examples/gnt-config-backup.in > doc/examples/gnt-config-backup
sed -f autotools/replace_vars.sed < doc/examples/hooks/ipsec.in > doc/examples/hooks/ipsec
sed -f autotools/replace_vars.sed < man/ganeti.html.in > man/ganeti.html
sed -f autotools/replace_vars.sed < man/ganeti-cleaner.html.in > man/ganeti-cleaner.html
sed -f autotools/replace_vars.sed < man/ganeti-confd.html.in > man/ganeti-confd.html
sed -f autotools/replace_vars.sed < man/ganeti-listrunner.html.in > man/ganeti-listrunner.html
sed -f autotools/replace_vars.sed < man/ganeti-masterd.html.in > man/ganeti-masterd.html
sed -f autotools/replace_vars.sed < man/ganeti-noded.html.in > man/ganeti-noded.html
sed -f autotools/replace_vars.sed < man/ganeti-os-interface.html.in > man/ganeti-os-interface.html
sed -f autotools/replace_vars.sed < man/ganeti-rapi.html.in > man/ganeti-rapi.html
sed -f autotools/replace_vars.sed < man/ganeti-watcher.html.in > man/ganeti-watcher.html
sed -f autotools/replace_vars.sed < man/gnt-backup.html.in > man/gnt-backup.html
sed -f autotools/replace_vars.sed < man/gnt-cluster.html.in > man/gnt-cluster.html
sed -f autotools/replace_vars.sed < man/gnt-debug.html.in > man/gnt-debug.html
sed -f autotools/replace_vars.sed < man/gnt-group.html.in > man/gnt-group.html
sed -f autotools/replace_vars.sed < man/gnt-instance.html.in > man/gnt-instance.html
sed -f autotools/replace_vars.sed < man/gnt-job.html.in > man/gnt-job.html
sed -f autotools/replace_vars.sed < man/gnt-node.html.in > man/gnt-node.html
sed -f autotools/replace_vars.sed < man/gnt-os.html.in > man/gnt-os.html
make[1]: Leaving directory `/home/loic/software/ganeti/ganeti-2.4.2'

```
Originally added on 2011-06-21 13:01:21 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The automake/build system is complex enough that solving this would take some effort. I think the rule here is that you shouldn't run the unittests on a host that already has Ganeti installed; we should document this, but otherwise we'll leave it as is.
```
Originally added on 2012-07-19 14:01:24 +0000 UTC.

Added Labels: Type-Documentation

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-30 15:59:14 +0000 UTC.

Changed State: Started
Added to Milestone: Release2.6

**Comment by Ganeti-Issues-Migrator** [comment]
```
I've updated the documentation slightly, and explained that this is an issue (and why). Will be in doc/devnotes.rst in 2.6.2.
```
Originally added on 2012-10-30 19:02:13 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking as released the bugs fixed in 2.6.2.
```
Originally added on 2013-01-11 12:13:05 +0000 UTC.

Changed State: Released
